### PR TITLE
Divide and Conquer

### PR DESCRIPTION
**Pull request by shellhub**
Find max and min value using Divide and Conquer algorithms

### CODE REVIEW

File: maths/find_max_recursion.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Divide and Conquer algorithm
+def find_max(nums, left, right):
+    """
+    find max value in list
+    :param nums: contains elements
+    :param left: index of first element
+    :param right: index of last element
+    :return: min in nums
```
Comment (cclauss) [inline maths/find_max_recursion.py]: :return: says min but should say max.

Also, add doctests as discussed in CONTRIBUTING.md
```python
>>> nums = [1, 3, 5, 7, 9, 2, 4, 6, 8, 10]
>>> find_max(nums, 0, len(nums) - 1) = max(nums)
True
```

Comment (shellhub) [reply maths/find_max_recursion.py]: Updated, Thanks for your review! @cclauss

Comment (cclauss) [reply maths/find_max_recursion.py]: Both files need doctests like the one above.

Comment (shellhub) [reply maths/find_max_recursion.py]: Please review again!

File: maths/find_max_recursion.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# Divide and Conquer algorithm
+def find_max(nums, left, right):
+    """
+    find max value in list
+    :param nums: contains elements
+    :param left: index of first element
+    :param right: index of last element
+    :return: max in nums
+
+    >>find_max([1, 3, 5, 7, 9, 2, 4, 6, 8, 10], 0, 9) == 10
```
Comment (cclauss) [inline maths/find_max_recursion.py]: greater than >, greater than >, greater than >, space

If not, it is not recognizing as a test.

File: maths/find_max_recursion.py (hunk 1 above)
Comment (shellhub) [reply maths/find_max_recursion.py]: > :return: says min but should say max.
> 
> Also, add doctests as discussed in CONTRIBUTING.md
> 
> ```python
> >>> nums = [1, 3, 5, 7, 9, 2, 4, 6, 8, 10]
> >>> find_max(nums, 0, len(nums) - 1) = max(nums)
> True
> ```

I update as you say, correct?

**Comment by shellhub** [comment]
Please review again @cclauss

File: maths/find_max_recursion.py (hunk 2 above)
Comment (shellhub) [reply maths/find_max_recursion.py]: you mean like this `nums = [9, 7, 5, 3, 2, 1]` ?

Comment (cclauss) [reply maths/find_max_recursion.py]: Nope.  I mean:
```python
  >>> find_max(...
# not...
  >>find_max(...
```

Comment (cclauss) [reply maths/find_max_recursion.py]: So that your tests appear here https://travis-ci.org/TheAlgorithms/Python/builds/595103352#L382

Comment (shellhub) [reply maths/find_max_recursion.py]: Thanks, I corrected follow as you say.

Comment (shellhub) [reply maths/find_max_recursion.py]: Please review again! @cclauss

File: maths/find_max_recursion.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+def find_max(nums, left, right):
+    """
+    find max value in list
+    :param nums: contains elements
+    :param left: index of first element
+    :param right: index of last element
+    :return: max in nums
+    """
+
+    """
```
Comment (cclauss) [inline maths/find_max_recursion.py]: Remove lines 9 and 11.

File: maths/find_min_recursion.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
+# Divide and Conquer algorithm
+def find_min(nums, left, right):
+    """
+    find min value in list
+    :param nums: contains elements
+    :param left: index of first element
+    :param right: index of last element
+    :return: min in nums
+    """
+
```
Comment (cclauss) [inline maths/find_min_recursion.py]: Remove lines 9 and 11.

**Comment by shellhub** [comment]
Please review again! @cclauss

**Comment by shellhub** [comment]
@cclauss Did you forget review?😹

File: maths/find_min_recursion.py (hunk 4 above)
Comment (cclauss) [reply maths/find_min_recursion.py]: Is this change done?

Comment (shellhub) [reply maths/find_min_recursion.py]: Yes!Done

**Pull request merged**